internal/librarian/generate.go

```diff
@@ -173,76 +182,67 @@ func (r *generateRunner) run(ctx context.Context) error {
 
 // generateSingleLibrary manages the generation of a single client library.
 //
-// It can either configure a new library if the API and library both are specified
-// and library not configured in state.yaml yet, or regenerate an existing library
-// if a libraryID is provided.
-// After ensuring the library is configured, it runs the generation and build commands.
-func (r *generateRunner) generateSingleLibrary(ctx context.Context, libraryID, outputDir string) error {
+// The single library generation executes as follows:
+//
+// 1. Configure the library, if the library is not configured in the state.yaml.
+//
+// 2. Generate the library.
+//
+// 3. Build the library.
+//
+// 4. Update the last generated commit.
+//
+// Returns the last generated commit *before* the generation and error, if any.
+func (r *generateRunner) generateSingleLibrary(ctx context.Context, libraryID, outputDir string) (string, error) {
 	if r.needsConfigure() {
 		slog.Info("library not configured, start initial configuration", "library", r.cfg.Library)
 		configuredLibraryID, err := r.runConfigureCommand(ctx)
 		if err != nil {
-			return err
+			return "", err
 		}
 		libraryID = configuredLibraryID
 	}
 
 	// At this point, we should have a library in the state.
 	libraryState := findLibraryByID(r.state, libraryID)
 	if libraryState == nil {
-		return fmt.Errorf("library %q not configured yet, generation stopped", libraryID)
+		return "", fmt.Errorf("library %q not configured yet, generation stopped", libraryID)
 	}
+	lastGenCommit := libraryState.LastGeneratedCommit
 
 	if len(libraryState.APIs) == 0 {
 		slog.Info("library has no APIs; skipping generation", "library", libraryID)
-		return nil
+		return "", nil
 	}
 
 	// For each library, create a separate output directory. This avoids
 	// libraries interfering with each other, and makes it easier to see what
 	// was generated for each library when debugging.
 	libraryOutputDir := filepath.Join(outputDir, libraryID)
 	if err := os.MkdirAll(libraryOutputDir, 0755); err != nil {
-		return err
+		return "", err
 	}
 
 	generatedLibraryID, err := r.runGenerateCommand(ctx, libraryID, libraryOutputDir)
 	if err != nil {
-		return err
-	}
-
-	if err := r.updateChangesSinceLastGeneration(generatedLibraryID); err != nil {
-		return err
+		return "", err
 	}
 
 	if err := r.runBuildCommand(ctx, generatedLibraryID); err != nil {
-		return err
+		return "", err
 	}
+
 	if err := r.updateLastGeneratedCommitState(generatedLibraryID); err != nil {
-		return err
+		return "", err
 	}
-	return nil
+
+	return lastGenCommit, nil
 }
 
 func (r *generateRunner) needsConfigure() bool {
 	return r.cfg.API != "" && r.cfg.Library != "" && findLibraryByID(r.state, r.cfg.Library) == nil
 }
 
-func (r *generateRunner) updateChangesSinceLastGeneration(libraryID string) error {
```

Comment:
Remove this function since commit history is not required when generating a library.